index.js

```diff
@@ -18,6 +20,22 @@ class Jarvis {
     this.constants = {}; // registered constants
   }
 
+  /**
+   * Checks for available scripts to switch mode to script mode if a
+   * script with specified extension is provided
+   */
+  async addScriptMode(extension) {
```

Comment:
where does this get called?
also, it's better for a function to receive the arguments than reaching for them from the inside (e.g. this function should not try to access process.argv on its own, right?)

Reply:
This is called in a third party CLI app which uses jarvis to enable the script mode.

ex :-
```
async () => {
      //adding script mode
      const res = await app.addScriptMode('kasaya', process.argv[2]);
      if (res) {
          process.exit();
      } else {
          const rl = Terminal.getTerminal();
          rl.prompt();      
      }
  });
```
This suggests that the CLI app accepts scripts with '.kasaya' extension...

Reply:
Agreed, added changes to the function to accept script as a parameter as well.